### DISCUSSION TITLE
Allow setting vector elements simultaneously

### DISCUSSION THREAD

**Discussion by TechnoPorg** [godotengine/godot-proposals#4232] [Editor]
In Blender, the vector editor built into the user interface allows you to drag your cursor across 2 or 3 of the vector elements vertically, then move the cursor left or right and edit them all simultaneously:
![image](https://user-images.githubusercontent.com/69441745/158709554-12535257-a2ca-4eca-b10d-dd014d189484.png)
This is quite helpful when doing operations like scaling cubes or planes, where you want quickly scale something up and maintain the aspect ratio. I don't believe Godot currently supports this feature, but it would make working with vectors in the editor much easier.

This wouldn't work with the optional horizontal layout of vectors in the inspector (Defaults to false for vec2, true for vec3). This feature could be enabled only in a vertical layout, or horizontal layouts could be removed from the inspector.

Doesn't work:
![image](https://user-images.githubusercontent.com/69441745/158709897-fa2480bf-73e9-4367-9b59-d5cb4c6291d3.png)

Elements could be set simultaneously. This also looks nicer, in my opinion.
![image](https://user-images.githubusercontent.com/69441745/158709938-e5e838b3-a314-430f-a9ec-b219458a0fc9.png)

**Comment by Mickeon** [comment]
> This wouldn't work with the optional horizontal layout of vectors in the inspector

I don't see why not. The dragging direction would just have to be rotated to the side.

Although, perhaps, it would be more universally useful to allow selecting more than one field at the same time (maybe with CTRL) for features such as these.

**Reply by TechnoPorg** [reply]
> The dragging direction would just have to be rotated to the side.

I don't think this would make much sense, since vertical dragging to change values would be the opposite of what is done most everywhere else in the editor. I like the idea of using CTRL, though. That would be the same as Calinou's PR for #144, which would be definitely enough for my needs. Although, it wouldn't be a very discoverable feature for new users. KoBeWi's PR for #144 I'm not as fond of, but it would also be great.

**Comment by Calinou** [comment]
See also https://github.com/godotengine/godot-proposals/issues/144.

**Comment by TechnoPorg** [comment]
I don't know if it's possible, or desired, to close discussions, but for my purposes, https://github.com/godotengine/godot/pull/59125 should work nicely!
